File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/entity/Ticket.java
package at.ac.tuwien.sepm.groupphase.backend.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.JoinColumn;
import javax.persistence.FetchType;
import javax.persistence.CascadeType;

@Getter
@Setter
@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "tickets")
public class Ticket {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.MERGE)
    @JoinColumn(name = "performance_id", referencedColumnName = "id")
    private Performance performance;

    @Column(nullable = false)
    private Double price;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnore
    private Booking booking;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "seat_id", referencedColumnName = "id")
    private Seat seat;

    @Column(nullable = false)
    private boolean isStandingTicket;
}

File: frontend/src/app/dtos/event/ticket.ts
import {Seat} from '../seat';
import {Performance} from './performance';

export class TicketToCreate {
  performanceId?: number;
  price: number;
  seat?: Seat;
  isStandingTicket: boolean;

  constructor(performanceId: number, price: number, seat: Seat, isStandingTicket: boolean) {
    this.performanceId = performanceId;
    this.price = price;
    if (seat) {
      this.seat = seat;
    }
    this.isStandingTicket = isStandingTicket;
  }

}

export class TicketShoppingCart {
  performance: Performance;
  price: number;
  seat?: Seat;
  isStandingTicket: boolean;

  constructor(performance: Performance, price: number, seat: Seat, isStandingTicket: boolean) {
    this.performance = performance;
    this.price = price;
    if (seat) {
      this.seat = seat;
    }
    this.isStandingTicket = isStandingTicket;
  }
}
